packages/docs-theme/src/components/documentation.tsx

```diff
@@ -349,7 +357,11 @@ export class Documentation extends React.PureComponent<DocumentationProps, Docum
         const { activeSectionId } = this.state;
         // only scroll nav menu if active item is not visible in viewport.
         // using activeSectionId so you can see the page title in nav (may not be visible in document).
-        const navItemElement = this.navElement.querySelector<HTMLElement>(`a[href="#${activeSectionId}"]`)!;
+        const navItemElement = this.navElement.querySelector<HTMLElement>(`a[href="#${activeSectionId}"]`);
+        if (navItemElement == null) {
+            return;
```

Comment:
this is what was actually crashing, and is all that is needed to avoid crashing. but without the rest of this diff users end up here which isn't really useful/seems like a bug on it's own

![Screenshot 2024-06-24 at 2 40 15 PM](https://github.com/palantir/blueprint/assets/14102129/1616c768-fbdf-4d16-acde-8101a961604c)

Reply:
Agree!